### ISSUE TITLE
Becoming Xenobio changeling doesn't carry over roundstart quirks/traits

### ISSUE THREAD

**Issue by Arkvold** [tgstation/tgstation#40040]
Issue reported from a round on a personal server.

Reporting client version: 512

When one becomes a Xenobio changeling (by using a Consciousness Transference Potion on a changeling headslug produced by plasma/blood gold slime reactions, then implanting your previous body with its egg and hatching), your roundstart quirks or traits are not preserved.  One would expect most of them to, as you're the same mind, complete with memories, in a fundamentally-identical body.

If one is habitually light on their feet (the Light Step quirk), they're not going to suddenly forget or stop just because they've transferred out of their original body and back into a biologically-identical copy of it.  Certain quirks I could see losing in the transition (prosthetic limb, for instance), since those are based on physical changes to the body itself, but quirks and traits associated with the mind should carry over to every body you inhabit, provided you do not lose your memories in the transition (such as dying and coming back as a golem).

This is largely a consistency issue, and may be broader than just Xenobio changelings.  Further testing may be necessary to find out under what circumstances one would lose one's roundstart traits and quirks inappropriately.

### Reproduction Steps:

1. Obtain Consciousness Transference Potion by using blood on a Rainbow Slime extract.
2. Spawn NPC changeling headslug by using blood on a Gold Slime extract
3. Use Consciousness Transference Potion on the headslug
4. Now in control of the headslug, approach and click on your former body to bite it and implant your egg; your headslug body will die, but this is of no consequence
5. Wait until you burst from the carcass in monkey form.
6. Use Human Form ability from Changeling menu.
7. Use Transform ability from the Changeling menu, and choose "drop flesh disguise."
8. **Problem Behavior:** _Although you are now in control of a body nearly identical to your previous one, with your mind and memories completely intact despite having briefly been a headslug and a monkey, any mind-associated quirks or roundstart traits you possessed, such as Light Step or Heavy Sleeper, are now gone._

**Comment by PKPenguin321** [comment]
Makes sense to me. You're completely abandoning your old body, so it would make sense to leave associated traits behind.

**Comment by RedFoxIV** [comment]
what about the pacifism one?

**Comment by Arkvold** [comment]
You're abandoning your old body, but not your old _mind_.  In the case of Xenobio changelings, the mind (memories, job role, even antag status) are all preserved throughout the procedure, and the resulting body is about as physically identical to your old one as it can be given your new species, right down to its genetics.

Shouldn't that also preserve mental quirks, such as Light Step, Pacifism, Heavy Sleeper, Anxiety, Depression, Heirloom (you don't necessarily shed sentimental attachment to a particular item), etc.?  Y'know, stuff that represents personal habits and personality.  I could understand losing Prosthetic Limb (the new body is essentially a changeling clone of your original, no reason for its limbs to not be intact) and maybe Nearsightedness (if it's not genetic, correcting one's vision would be a trivial exercise of changeling shapeshifting, and even if it was, changelings can control that too), but I don't see why they'd lose other traits.

Doing some further testing, the whole 'leaving your body behind' argument doesn't quite work here - there are transformations that very much _don't_ leave your body behind, yet also kill your quirks.  Examples include being monkeyed and then changed back by genetics using a UI+UE+SE made from your own DNA prior to monkeying (you are literally the same person as before, down to the genetics), drinking the bottle of dragon's blood and being turned into a lizard or skeleton (you're the same person, just now bearing scales and a tail or significantly less flesh), or being turned into a jellyperson (the only difference between you then and you now is the application of some slime mutagen).

Just to make it clear, **I'm not arguing just to keep the good mental traits.**  Whether they're good (Skittish) or bad (Pacifism), I all mind-related quirks and traits should be passed over on a mind-retaining transformation; they're part of _who_ you are, not _what_ you are.  For example, If you have an heirloom that means so much to you that you're depressed without it, it shouldn't lose its sentimental value just because you died and came back after being harvested out of a Replica pod.  After all, if you can remember what your name, rank, and job is, you ought to remember what's sentimental to you.

**Comment by timothymtorres** [comment]
This is a big problem with shapeshifting and polymorphing code.  Linking the following issues here:  #35539, #42308, #59127
